provider/server/machine_create_test.go

```diff
@@ -4,17 +4,21 @@
 package server_test
 
 import (
+	"github.com/digitalocean/go-libvirt"
 	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
 	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("CreateMachine", func() {
 
-	It("should correctly create a machine", func(ctx SpecContext) {
+	It("should create a machine", func(ctx SpecContext) {
 		By("creating a machine")
-		res, err := srv.CreateMachine(ctx, &iri.CreateMachineRequest{
+		ignitionData := []byte("urjhikmnbdjfkknhhdddeee")
+		imageURL := "ghcr.io/ironcore-dev/ironcore-image/gardenlinux:rootfs-dev-20231206-v1"
+		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
```

Comment:
Similarly, the machine creation can be extracted into a separate function for code reuse.

Reply:
The direct use of `machineClient.CreateMachine` in the test ensures clear and straightforward readability. Adding another function may introduce unnecessary abstraction without significant improvements in clarity.

Reply:
I suggest reusing the code instead of duplicating it across multiple places. Providing a meaningful name will enhance code clarity. While utilizing machineClient.CreateMachine directly aligns with your reasons, it is advisable to centralize the machine definition in one location for broader usage.